### ISSUE TITLE
mask_overlap=False still gives overlapping masks

### ISSUE THREAD

**Issue by polinamalova0** [ultralytics/ultralytics#13487]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train, Predict

### Bug

I am training the model and have overlap_mask=False in both training and prediction commands. however, the output shows overlapping masks. I checked, there is no overlapping polygons in the annotated images. 

### Environment

Ultralytics YOLOv8.2.16 �🚀 Python-3.10.4 torch-2.3.0+cu121 CUDA:0 (NVIDIA GeForce RTX 2070, 8192Mi
Setup complete ✅ (16 CPUs, 15.6 GB RAM, 308.4/473.8 GB disk)

### Minimal Reproducible Example
```
from ultralytics import YOLO
import torch

device = 'cuda' if torch.cuda.is_available() else 'cpu'
print(f'Using device: {device}')

model = YOLO('yolov8s-seg.pt').to(device)
if __name__ == '__main__':
    results = model.train(data=r'\dataset.yaml', epochs=100,  imgsz=(428,428), device = device, overlap_mask=False, verbose=True, workers = 1)

path = r'\train\weights\last.pt'
img = r'foto.png'
model = YOLO(path)
prediction = model(img, save=True, overlap_mask=False, verbose =True)
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@polinamalova0 hi there,

Thank you for bringing this to our attention! It looks like you've provided a good start with your minimal reproducible example. However, to ensure we can fully investigate this issue, could you please confirm the following:

1. **Latest Versions**: Ensure you are using the latest versions of both `torch` and `ultralytics`. You can update them using:
    ```bash
    pip install --upgrade torch ultralytics
    ```

2. **Minimal Reproducible Example**: Your provided code is a great start. Could you also share a sample of your dataset or a few annotated images where this issue occurs? This will help us reproduce the problem more accurately. For more details on creating a minimal reproducible example, please refer to our [guide](https://docs.ultralytics.com/help/minimum_reproducible_example).

3. **Code Example**: Here’s a slightly modified version of your code to ensure `overlap_mask=False` is correctly set during both training and prediction:
    ```python
    from ultralytics import YOLO
    import torch

    device = 'cuda' if torch.cuda.is_available() else 'cpu'
    print(f'Using device: {device}')

    model = YOLO('yolov8s-seg.pt').to(device)
    if __name__ == '__main__':
        results = model.train(data='dataset.yaml', epochs=100, imgsz=(428, 428), device=device, overlap_mask=False, verbose=True, workers=1)

    path = 'train/weights/last.pt'
    img = 'foto.png'
    model = YOLO(path)
    prediction = model(img, save=True, overlap_mask=False, verbose=True)
    ```

Please make sure the `overlap_mask` parameter is correctly applied in both the training and prediction phases.

Once you’ve confirmed the above steps, if the issue persists, we’ll be happy to dive deeper into it. Your cooperation helps us improve YOLOv8 for everyone in the community! 🚀
